app/controllers/listings_controller.rb

```diff
@@ -124,7 +124,23 @@ def process_unsuccessful_update
   end
 
   def process_after_update
-    redirect_to "/listings"
+    # The following sets variables used in the index view. We render the
+    # index view directly to avoid having to redirect.
+    #
+    # Redirects lead to a race condition where we redirect to a cached view
+    # after updating data and we don't bust the cache fast enough before
+    # hitting the view, therefore stale content ends up being served from
+    # cache.
+    #
+    # https://github.com/forem/forem/issues/10338#issuecomment-693401481
+    @listings = Listing.where(published: true)
```

Comment:
Might be nice to take this "default scope" for listings which is also used in the `index` view and create a method for it. That way if we want to change the index view we dont forget about this straggler down here.

Reply:
Great idea! Updated!